wayland-backend/src/protocol.rs

```diff
@@ -2,34 +2,38 @@ use std::{ffi::CString, os::unix::io::RawFd};
 
 pub use wayland_sys::common::{wl_argument, wl_interface, wl_message};
 
+/// Describes whether an argument may have a null value.
 #[derive(Clone, Copy, PartialEq, Eq, Debug)]
 pub enum AllowNull {
+    /// Null values are allowed.
     Yes,
+    /// Null values are forbidden.
     No,
 }
 
 /// Enum of possible argument types as recognized by the wire
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
 pub enum ArgumentType {
-    /// i32
+    /// An integer argument. Represented by a [`i32`].
     Int,
-    /// u32
+    /// An unsigned integer argument. Represented by a [`u32`].
     Uint,
-    /// fixed point, 1/256 precision
+    /// A fixed point, 1/256 precision signed floating point number.
```

Comment:
```suggestion
    /// A signed fixed point number with 1/256 precision
```

If it is a fixed point number, then by definition it is not a floating point number. ^^